web-console/src/lib/functions/common/tanstack.ts

```diff
@@ -23,22 +25,59 @@ type QueryType<U extends Record<string, FunctionType>, P extends keyof U> = {
  * @returns
  */
 export const mkQuery = <U extends Record<string, FunctionType>>(
-  source: U
+  source: U,
+  options?: Partial<
+    Omit<QueryType<U, keyof U>, 'queryKey' | 'qyeryFn'> & {
+      wrapQueryFn: <T extends any[], R extends Promise<any>>(f: (...args: T) => R) => (...args: T) => R
+    }
+  >
 ): { [P in keyof U]: (...args: Arguments<U[P]>) => QueryType<U, P> } =>
   Object.fromEntries(
     Object.entries(source).map(([key, value]) => {
+      const wrapper = options?.wrapQueryFn ?? identity
       return [
         key,
         (...args: unknown[]) => {
           return {
             queryKey: [key, ...args],
-            queryFn: () => value(...args)
+            queryFn: () => wrapper(value)(...args),
+            ...(options ?? {})
           }
         }
       ]
     })
   ) as any
 
+/**
+ * Construct a set of objects with queryKey,
+ * bound to the same names as the fields of an argument object
+ *
+ * Fields of source object become a prefix for a corresponding queryKey
+ * Functions' arguments are prepended as-is in a list to the queryKey
+ * @param source
+ * @returns
+ */
+export const mkQueryKey = <U extends Record<string, FunctionType>>(
+  source: U
+): { [P in keyof U]: (...args: Arguments<U[P]>) => QueryType<U, P> } =>
```

Comment:
It is :) This is a complex helper, but it automatically deduces query string identifier (that is used as a query cache key), argument and return types of a query from a simple definition at pipelineManagerQuery.ts#L97 .
It enables syntax `useQuery(PipelineManagerQuery.programCode(programName))` instead of 
`useQuery<ProgramDescr>({
queryKey: ["programCode", programName],
})` and corresponding `queryFn: (...args) => ProgramService.getProgram(args[0], true)` defined elsewhere, and having to make sure correct arguments and type is specified at call site